File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> memo(n + 1, 0);
        for (int i = 1; i <= n; ++i) {
            memo[i] = memo[i >> 1] + (i & 1);
        }
        return memo;
    }
};

File: 0338-counting-bits/0338-counting-bits.py
class Solution(object):
    def countBits(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        
        arr = [0] * (n + 1)
        for x in range(1, n + 1):
            arr[x] = arr[x & (x - 1)] + 1

        return arr